framework_lib/src/smbios.rs

```diff
@@ -351,3 +353,107 @@ fn kenv_get(name: &str) -> nix::Result<String> {
         Ok(value)
     }
 }
+
+#[derive(Debug)]
+enum SmbiosSerialNumber {
+    Mainboard = 1,
+    Laptop,
+    Camera,
+    Display,
+    Battery,
+    Touchpad,
+    Keyboard,
+    Fingerprint,
+    AudioDaughtercard,
+    ACover,
+    BCover,
+    CCover,
+    AntennaMain,
+    AntennaAux,
+    TouchpadFpc,
+    FingerprintFfc,
+    EdpCable,
+    LcdCable,
+    ThermalAssembly,
+    WifiModule,
+    Speaker,
+    RamSlot1,
+    RamSlot2,
+    Ssd,
+    AudioFfc,
+}
+
+pub fn dump_oem_strings(strings: &SMBiosStringSet) {
+    for (i, s) in strings.into_iter().enumerate() {
+        let idx = i + 1;
+        let sn = match idx {
```

Comment:
It would be cooler if you didn't use idx positions for decoding the serial number (remember what happened last time these positions changed?).  Just put them in a big array, shuffle them, and try to tell what each of them is based on the serial number contents.

Reply:
The order doesn't change, the factory also depends on it, as far as I remember.

Decoding the serialnumber is harder as we would have to embed all possible serialnumbers and I couldn't include those for unannounced projects.